source/plugins/ovs-plugin.rst

```diff
@@ -364,6 +364,47 @@ To enable DPDK on VM deployments:
       dpdk-interface-model:
       <model type='virtio'/>
 
+Additional configurations on service offerings
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It is also possible to avoid passing additional configuration on each VM deployment, but setting these configurations on a service offering, and those are passed to the VM. To create a service offering with additional configurations, pass each key/value pair as service offering details on service offering creation, with keys starting with the "extraconfig" keyword, and each value an URL UTF-8 encoded string. For example:
+
+::
+   
+   create serviceoffering name=<NAME> displaytext=<NAME> serviceofferingdetails[0].key=extraconfig-dpdk-hugepages serviceofferingdetails[0].value=%3CmemoryBacking%3E%20%3Chugepages%2F%3E%20%3C%2FmemoryBacking%3E serviceofferingdetails[1].key=extraconfig-dpdk-numa serviceofferingdetails[1].value=%3Ccpu%20mode%3D%22host-passthrough%22%3E%20%3Cnuma%3E%20%3Ccell%20id%3D%220%22%20cpus%3D%220%22%20memory%3D%229437184%22%20unit%3D%22KiB%22%20memAccess%3D%22shared%22%2F%3E%20%3C%2Fnuma%3E%20%3C%2Fcpu%3E
+
+Additional configurations are stored as service offering details.
+
+DPDK vHost User mode selection
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The vHost user mode describes a client/server model between Openvswitch along with DPDK and QEMU, in which one acts as client while the other as server. The server creates and manages the vHost user sockets and the client connects to the sockets created by the server:
+
+- DPDK vHost user server mode:
+   - Is the default configuration.
+   - OVS with DPDK acts as the server, while QEMU acts as the client.
+   - The port types used are: dpdkvhostuser
+
+- DPDK vHost user client mode:
+   - OVS with DPDK acts as the client and QEMU acts as the server.
+   - If Openvswitch is restarted then the sockets can reconnect to the existing sockets on the server, and normal connectivity can be resumed.
+   - The port types used are: dpdkvhostuserclient
+
+It is possible to pass the DPDK vHost User mode as a service offering detail with key: "DPDK-VHOSTUSER" and values: "client" or "server". The following table illustrates the expected behaviour on DPDK ports and VM guest interfaces.
```

Comment:
Can this only be passed via a service offering ?

Maybe the Mode options should be explained before the ability to embed dpdk in the service offerings. And then the ability to use service offerings explained separately

Reply:
Refactored